Formula/haskell-stack.rb

```diff
@@ -1,10 +1,33 @@
 class HaskellStack < Formula
   desc "Cross-platform program for developing Haskell projects"
   homepage "https://haskellstack.org/"
-  url "https://github.com/commercialhaskell/stack/archive/v2.7.3.tar.gz"
-  sha256 "37f4bc0177534782609ec3a67ec413548d3f2cabff7c4c0bc8a92a36e49c6877"
   license "BSD-3-Clause"
-  head "https://github.com/commercialhaskell/stack.git", branch: "master"
+
+  stable do
+    url "https://github.com/commercialhaskell/stack/archive/v2.7.3.tar.gz"
+    sha256 "37f4bc0177534782609ec3a67ec413548d3f2cabff7c4c0bc8a92a36e49c6877"
+
+    # - Fixed building from source.
+    #
+    #   Due to recent update of aeson-2.0.0.0, stack can no longer be built
+    #   with cabal-install. So I patched stack to freeze cabal dependencies
+    #   using stackage 17.15 LTS.
+    #
+    # - Add Apple Silicon support.
+    #
+    #   HEAD version of stack has already been patched to support Apple
+    #   Silicon. However, the next release containing that patch hasn't
+    #   release yet. So I manually patched stack v2.7.3 to support Apple
+    #   Silicon.
+    #
+    # Reference:
+    #   https://github.com/commercialhaskell/stack/pull/5562
+    #   https://github.com/commercialhaskell/stack/pull/5677
+    patch do
+      url "https://github.com/commercialhaskell/stack/compare/7927a3ae..32b80476.patch?full_index=1"
+      sha256 "8e87e3f3409cf1427b46124586536c91f94dde4c27d2708764604c24c4ea98d7"
+    end
```

Comment:
This seems like a pretty big patch to apply unconditionally. Can we use a smaller patch? We should also use the commits from the relevant pull requests instead of doing `compare/<hash>..<hash>`.

Reply:
That patch does only two thing.

1. First commit: Let “cabal build” get the exactly same result with “stack build”
2. Second commit: Add M1 support by adding only a single line `Platform AArch64               Cabal.OSX     -> return "macosx-aarch64"`

This is the minimalist patch that I could made. I’ll accept feedback if there is another way to do better or smaller.

Reply:
I tried to make a smaller patch in [this way](https://github.com/simnalamburt/stack/compare/v2.7.3..no-aeson-2), but I found that another dependency has broken their API as well.

```
src/Stack/Setup.hs:847:24: error:
    • Couldn't match expected type ‘SimpleRepo’ with actual type ‘Repo’
    • In the first argument of ‘withRepo’, namely ‘repo’
      In the expression: withRepo repo
      In a stmt of a 'do' block:
        withRepo repo
          $ do mcwd <- traverse parseAbsDir =<< view workingDirL
               let cwd = fromMaybe (error "Invalid working directory") mcwd
               threads <- view $ configL . to configJobs
               let hadrianArgs = fmap T.unpack ...
                   hadrianScripts
                     | osIsWindows = hadrianScriptsWindows
                     | otherwise = hadrianScriptsPosix
               ....
    |
847 |        Pantry.withRepo repo $ do
    |                        ^^^^
```

Reply:
Ok, that's fine, but we should still do this:

> We should also use the commits from the relevant pull requests instead of doing `compare/<hash>..<hash>`.

Reply:
Thanks @carlocab! But.. actually I couldn't understand what you meant by "use the commits from the relevant pull requests instead of doing compare/<hash>..<hash>.". Is there any special Homebrew DSL for that? (Sorry this is my first time submitting patch to the Homebrew)

Reply:
Oh did you mean that there is a way to apply multiple patches at once?

Reply:
You can have multiple patch blocks. But you should have the URL of a single commit per block, not the `aaa...bbb` style url.

Reply:
Thanks for the help! I've updated the PR :) 🎉